Outline.md

```diff
@@ -0,0 +1,137 @@
+    4.  Install lowdb (json database) (screen)
+    5.  Install Necessary Middleware
+4.  Install and Explore Postman
+    1.  Download and Install Postman (screen)
+    2.  Overview of Postman (screen)
+5.  Create Our First REST API
+    1.  Explain The Todo Application (screen)
+    2.  Create A POST Route (screen)
+    3.  Create GET Routes (screen)
+    4.  Create PUT & PATCH Routes (screen)
+    5.  Create DELETE Route (screen)
+6.  CRUD Operations for Recipe
+    1.  Setup Database (Screen)
+    2.  Implement Create Recipe (Screen)
+    3.  Implement Get Recipes (Screen)
+    4.  Implement Get Recipe By Id (Screen)
+    5.  Implement Update Recipe (Screen)
+    6.  Implement Delete Recipe (Screen)
+7.  CRUD Operation for Review
+    1.  Understand The Relation Between Recipe and Review (Screen)
+    2.  Implement Create Review (Screen)
+    3.  Implement Get Review (Screen)
+    4.  Implement Get Review By Id (Screen)
+    5.  Implement Update Review (Screen)
+    6.  Implement Delete Review (Screen)
+8.  Understand JWT & Authentication
+    1.  How to Protect API (Screen)
+    2.  What is JWT (Screen)
+    3.  Structure of JWT (Screen)
+    4.  How Does JWT Authentication Work (Screen)
+    5.  Create User Model (Screen)
+    6.  Create Registration Routes (Screen)
+    7.  Create Login Routes (Screen)
+9.  Authentication using PassportJS
+    1.  Create Authentication Middleware (Screen)
+    2.  Understand PassportJS (Screen)
+    3.  Implement PassportJS Local Strategy (Screen)
+10.  REST API Best Practices
+     1.  Open API Specifications (Screen)
+     2.  API Best Practices (Slide)
+11.  API Versioning and Documentation
+     1.  How to Manage Version of API (Screen)
+     2.  Introduction to Swagger (Screen)
+     3.  Implement Swagger in Our Application (Screen)
+12.  API Testing using JEST & Supertest
+     1.  Understand Automated Testing (Slide)
+     2.  Install JEST & Supertest (Screen)
+     3.  Implement Testing for Create Recipe (Screen)
+     4.  Implement Testing for Update Recipe (Screen)
+     5.  Implement Testing for Find Recipes (Screen)
+13.  Create Our First GraphQL API
+     1.  Understand the Core Concept of GraphQL (Slide)
+     2.  Setup GraphQL Yoga (Screen)
+     3.  Understand Schema (Screen)
+     4.  Understand Resolvers (Screen)
+     5.  Explore The GraphQL Playground (Screen)
+14.  Create Resolvers for Recipe
+     1.  Create Schema for Recipe (Screen)
+     2.  Create Query Resolvers for Recipe (Screen)
+     3.  Create Mutation Resolvers for Recipe (Screen)
+15.  Create Resolvers for Review
+     1.  Create Schema for Review (Screen)
+     2.  Create Query Resolvers for Review (Screen)
+     3.  Create Mutation Resolvers for Review (Screen)
+16.  Add Authentication to GraphQL
+     1.  Authentication Technique for GraphQL (Screen)
+     2.  Implement PassportJS Local Authentication Strategy (Screen)
+17.  GraphQL Best Practices
+     1.  GraphQL Best Practices (Slide)
+18.  What Next
```

Comment:
Is there any other API technologies you think we should focus on here? I've seen gRPC becoming more and more popular for those who do everything on Google Cloud. 

Perhaps we can have a section on "up and coming technologies" that does a _very simple_ walkthrough through other API technologies - what do you think?

Reply:
sounds good. we can add that too.